### PR TITLE
feat: add node 12 support

### PR DESCRIPTION
**Pull request by shaimendel**
- [x] Ready for review
- [ ] Follows CONTRIBUTING rules
- [ ] Any changes involving paths support all path separators
- [ ] Reviewed by Snyk internal team

#### What does this PR do?
Adds support to Node 12 in the agent's tests

#### Where should the reviewer start?
commit by commit

note for reviewer: didn't add Windows support for Node 12 because I couldn't fix that race condition there

### CODE REVIEW

File: test/e2e.test.js [hunk 1]
```diff
@@ -167,7 +167,7 @@ test('demo app reports a vuln method when called', async (t) => {
 
   // wait until we refresh the snapshot
   await snapshotSentPromise;
-
+  await sleep(100); // race in node 12
```
Comment (Shesekino) [inline test/e2e.test.js:170]: what race?

File: test/e2e.test.js [hunk 2]
```diff
@@ -6,13 +6,16 @@ const path = require('path');
 
 test('demo app reports a vuln method when called', async (t) => {
   const newSnapshotModificationDate = new Date();
-
+  let snapshotSentResolve;
+  const snapshotSentPromise = new Promise(resolve => {
```
Comment (Shesekino) [inline test/e2e.test.js:10]: how is that related to undoing unnecessary JSON parse?

File: .travis.yml [hunk 3]
```diff
@@ -17,7 +17,7 @@ cache:
   directories:
     - node_modules
 install:
-  - npm install
+  - npm ci
```
Comment (Shesekino) [inline .travis.yml:20]: I'm missing the "why" here in the commit description

File: .travis.yml [hunk 4]
```diff
@@ -12,6 +12,7 @@ language: node_js
 node_js:
   - '8'
   - '10'
+  - '12'
```
Comment (Shesekino) [inline .travis.yml:15]: needs appveyor as well? or alternatively change the commit description to say we only support node 12 on linux/mac?

Comment (Shesekino) [reply .travis.yml:15]: also not sure why that's a ```feat```

File: .travis.yml (hunk 3 above)
Comment (FauxFaux) [reply .travis.yml:20]: I don't think this should even have a package-lock, we're consumed mostly as a library, right?

**Comment by shaimendel** [comment]
closing for now, until we have time to handle it
